File: android/src/main/java/com/margelo/nitro/mqtt_lib/HybridMQTT.kt
package com.margelo.nitro.nitroMath

import android.util.Log
import com.margelo.nitro.mqtt_lib.HybridMQTTSpec

class HybridMQTT : HybridMQTTSpec() {
    override val memorySize: Long
        get() = 0L

    override fun connect(host: String, port: Double): String {
        Log.d("connect","Connecting to MQTT server: ${host} - ${port}")
        return "Connected".toString()
    }

    override fun disconnect() {
         Log.d("disconnect","disconnect")
    }

    override fun subscribe(topic: String, callBack: (data: String) -> Unit) {
        Log.d("subscribe","Subscribing to topic: ${topic}")
    }

    override fun unsubscribe(topic: String) {
        Log.d("unsubscribe","unsubscribe to topic: ${topic}")
    }

    override fun publish(topic: String, message: String): String {
        Log.d("Publishing","Publishing message: ${topic} - ${message}")
        return "publish"
    }
}

File: ios/HybridMQTT.swift
// import CocoaMQTT5

class HybridMQTT : HybridMQTTSpec {
    ///MQTT 5.0

  // Properties
    public var hybridContext = margelo.nitro.HybridContext()
    public var memorySize: Int {
       return getSizeOf(self)
    }

  // Methods
    public func connect(host: String, port: Double) throws -> String {
        print("connect", host, port)
        return "connected"
   }
   public func disconnect() throws -> Void {
        print("disconnect")
   }
   public func subscribe(topic: String) throws -> Void {
        print("subscribe", topic)
   }

   public func unsubscribe(topic: String) throws -> Void {
        print("unsubscribe", topic)
   }
   public func publish(topic: String, message: String) throws -> Void {
        print("publish", topic, message)
   }
   public func subscribe(topic: String, callBack: @escaping ((String) -> Void)) throws {
       print("subscribe", topic)
    }
    
    public func publish(topic: String, message: String) throws -> String {
        print("publish", topic)
        return "published"
    }

}
